pkg/workloads/console/events/gcp_pubsub.go

```diff
@@ -0,0 +1,69 @@
+package events
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"cloud.google.com/go/pubsub"
+)
+
+type ErrorNoTopic struct {
+	Topic string
+}
+
+func (e ErrorNoTopic) Error() string {
+	return fmt.Sprintf("No topic found with id: %s", e.Topic)
+}
+
+var _ error = &ErrorNoTopic{}
+
+type GooglePubSubPublisher struct {
+	client *pubsub.Client
+	topic  *pubsub.Topic
+}
+
+var _ Publisher = &GooglePubSubPublisher{}
+
+func NewGooglePubSubPublisher(ctx context.Context, projectName string, topicName string) (*GooglePubSubPublisher, error) {
+	client, err := pubsub.NewClient(ctx, projectName)
+	if err != nil {
+		return nil, err
+	}
+
+	return &GooglePubSubPublisher{
+		client: client,
+		topic:  client.Topic(topicName),
+	}, nil
+}
+
+func (p *GooglePubSubPublisher) Stop() {
+	p.topic.Stop()
+}
+
+func (p *GooglePubSubPublisher) Publish(ctx context.Context, msg interface{}) (string, error) {
+	messageBytes, err := json.Marshal(msg)
+	if err != nil {
+		return "", err
+	}
+
+	result := p.topic.Publish(ctx, &pubsub.Message{Data: messageBytes})
+	id, err := result.Get(ctx)
+	if err != nil {
+		return "", err
+	}
+	return id, nil
+}
+
+func (p *GooglePubSubPublisher) Health(ctx context.Context) error {
```

Comment:
What uses this?

Reply:
Nothing uses this yet, and I'm happy to omit this from this changeset to reduce the scope of the changes.
This was more around trying to build a nice health reporting standard to our theatre operators, and using this changeset as a place to experiment.

Reply:
Yeah I think lets omit from this.